File: ruby/clock/clock.rb
class Clock
  attr_reader :hours, :minutes

  def initialize(hours, minutes)
    @minutes = minutes % 60
    @hours = (hours + minutes / 60) % 24
  end

  def self.at(hours, minutes)
    new(hours, minutes)
  end

  def to_s
    format('%02d:%02d', hours, minutes)
  end

  def +(mins)
    self.class.new(hours, minutes + mins)
  end

  def -(mins)
    self.class.new(@hours, @minutes - mins)
  end

  def ==(other_clock)
    hours == other_clock.hours && minutes == other_clock.minutes
  end
end

module BookKeeping
  VERSION = 2
end

File: python/clock/clock.py
class Clock(object):
    """A class that handles military time."""

    def __init__(self, hours, minutes):
        self.minutes = minutes % 60
        self.hours = (hours + (minutes // 60)) % 24

    def __repr__(self):
        return '{classname}(hours={hours}, minutes={minutes})'.format(
            classname=self.__class__.__name__,
            hours=self.hours,
            minutes=self.minutes)

    def __str__(self):
        return '{:02d}:{:02d}'.format(self.hours, self.minutes) 

    def __eq__(self, other):
        return self.hours == other.hours and self.minutes == other.minutes

    def __add__(self, minutes):
        return Clock(self.hours, self.minutes + minutes)

    def __sub__(self, minutes):
        return Clock(self.hours, self.minutes - minutes)
